support/static/scripts/payments.js

```diff
@@ -145,35 +146,55 @@
     requestPayerEmail: true
   });
 
-  // Callback when a payment method is created.
-  paymentRequest.on('paymentmethod', async event => {
-    // Confirm the PaymentIntent with the payment method returned from the payment request.
-    const {error} = await stripe.confirmPaymentIntent(
-      paymentIntent.client_secret,
-      {
-        payment_method: event.paymentMethod.id
-      }
-    );
-    if (error) {
-      // Report to the browser that the payment failed.
-      event.complete('fail');
-      handlePayment({error});
+  function handleServerResponse(response, complete) {
+    if (!complete) {
+      complete = str => console.log(str);
+    }
+    if (response.error) {
+      complete("fail");
+    } else if (response.requires_action) {
+      complete("success");
+      handleAction(response);
     } else {
-      // Report to the browser that the confirmation was successful, prompting
-      // it to close the browser payment method collection interface.
-      event.complete('success');
-      // Let Stripe.js handle the rest of the payment flow, including 3D Secure if needed.
-      const response = await stripe.handleCardPayment(
-        paymentIntent.client_secret
-      );
-      handlePayment(response);
+      complete("success");
     }
-  });
+  }
+
+  function handleAction(response) {
+    stripe
+      .handleCardAction(response.payment_intent_client_secret)
+      .then(async function(result) {
```

Comment:
`then` here can also be replaced with `await` to avoid nesting, while `handleAction` itself can become `async`